docs/en/ingest-management/elastic-agent/configuration/outputs/output-shared-settings.asciidoc

```diff
@@ -0,0 +1,78 @@
+// These settings are shared across the docs for multiple inputs. Copy and use
+// the following template to add a shared setting. Replace values in all caps.
+// Use an include statement // to pull the tagged region into your source file:
+// include::output-shared-settings.asciidoc[tag=NAME-setting]
+
+
+// tag::NAME-setting[]
+|
+[id="input-{input-type}-NAME-setting"]
+`NAME`
+
+| (TYPE) DESCRIPTION.
+
+*Default:* `DEFAULT`
+
+OPTIONAL INFO AND EXAMPLE
+// end::NAME-setting[]
+
+// =============================================================================
+
+// tag::compression_level-setting[]
+|
+[id="output-{output-type}-compression_level-setting"]
+`compression_level`
+
+| (int) The gzip compression level. Set this value to `0` to disable compression.
+The compression level must be in the range of `1` (best speed) to `9` (best
+compression).
+
+Increasing the compression level reduces network usage but increases CPU usage.
+
+// Default varies by output
+// end::compression_level-setting[]
+
+// =============================================================================
+
+// tag::enabled-setting[]
+|
+[id="output-{output-type}-enabled-setting"]
+`enabled`
+
+| (boolean) Enables or disables the output. If set to `false`, the output is
+disabled.
+
+//The default varies by output type
```

Comment:
There must be some output types where this varies? I think I will leave off the default for the enabled setting because it seems like users who don't use the setting will expect the output to be "enabled" if it's present in the config.